### PR TITLE
Continue to Photoshop button added on Hue-Sat marquee.

### PR DESCRIPTION
**Pull request by suhjainadobe**
Functionality of continue to photoshop added on Hue-Sat marquee.

Resolves: [MWPW-142976](https://jira.corp.adobe.com/browse/MWPW-NUMBER)

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.live/drafts/suhjain/interactive-marquee/continuetops
- After: https://openinps--cc--adobecom.hlx.live/drafts/suhjain/interactive-marquee/continuetops

### CODE REVIEW

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 1]
```diff
@@ -173,7 +187,39 @@ function sliderEvent(media, layer) {
   });
 }
 
-function uploadImage(media, layer) {
+function cssToPhotoshop(imgObj, adjustment, value) {
+  const unitValue = convertToUnit(adjustment, value, CSSRanges);
+  imgObj[adjustment] = convertFromUnit(adjustment, unitValue, PsRanges);
+}
+
+function convertToUnit(adjustment, value, ranges) {
```
Comment (aishwaryamathuria) [inline creativecloud/features/interactive-components/slider-tray/slider-tray.js:195]: convertToUnit => convertToPSUnit

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 2]
```diff
@@ -173,7 +187,39 @@ function sliderEvent(media, layer) {
...
+  if (value < ranges[adjustment].min || value > ranges[adjustment].max) {
+    throw new Error(`value out of range ${adjustment}:${value}`);
```
Comment (aishwaryamathuria) [inline creativecloud/features/interactive-components/slider-tray/slider-tray.js:197]: Can we return a default here in place of throwing error?

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 3]
```diff
@@ -173,7 +187,39 @@ function sliderEvent(media, layer) {
...
+    const spread = ranges[adjustment].zero - ranges[adjustment].min;
+    return (value - ranges[adjustment].min) / spread - 1;
+  }
+  const spread = ranges[adjustment].max - ranges[adjustment].zero;
+  return (value - ranges[adjustment].zero) / spread;
+}
+
+function convertFromUnit(adjustment, value, ranges) {
+  if (value < -1 || value > 1) {
+    throw new Error(`value out of range ${adjustment}:${value}`);
```
Comment (aishwaryamathuria) [inline creativecloud/features/interactive-components/slider-tray/slider-tray.js:210]: Can we return a default here in place of throwing error?

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 4]
```diff
@@ -199,6 +247,52 @@ function uploadImage(media, layer) {
...
+      const psurls = [
+        'https://photoshop.adobe.com',
+        'https://dev.photoshop.adobe.com',
+        'https://pr.photoshop.adobe.com/?PR=48898',
+        'https://localhost.corp.adobe.com:3000',
+      ];
+      const { openInPsWeb } = await import('../../../deps/openInPsWeb/openInPsWeb.js');
+      const imageData = await (await fetch(imgObj.imgSrc)).blob();
+      openInPsWeb(
+        psurls[1],
```
Comment (aishwaryamathuria) [inline creativecloud/features/interactive-components/slider-tray/slider-tray.js:287]: Should be picked form PS env url value and not hard coded

**Review by aishwaryamathuria — CHANGES_REQUESTED** [review]

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js (hunk 1 above)
Comment (suhani17jain) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js:195]: This is not PS value, instead a conversion in between.

File: creativecloud/features/interactive-components/slider-tray/slider-tray.css [hunk 5]
```diff
@@ -322,40 +322,50 @@
    display: none;
  }
 
- .interactive-enabled .step-slider-tray .sliderTray .svg-icon-container {
+ .interactive-enabled .step-slider-tray .sliderTray .svg-icon-container img,
+ .interactive-enabled .step-slider-tray .continueButton .svg-icon-container img{
```
[the remark this answers is not in the document]
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.css]: done

File: creativecloud/features/interactive-components/slider-tray/slider-tray.css [hunk 6]
```diff
@@ -322,40 +322,50 @@
...
+@media screen and (min-width: 1024px) {
+  .interactive-enabled .step-slider-tray .continueButton {
+    position: absolute;
+    bottom: 3%;
+    display: flex;
+    gap: 16px;
+    background: #1273E6;
+    border-radius: 8px;
+    text-decoration: none;
+    color: var(--prompt-btn-fill-light) !important;
```
[the remark this answers is not in the document]
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.css]: done

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 7]
```diff
@@ -3,11 +3,23 @@
 import { createTag } from '../../../scripts/utils.js';
 import defineDeviceByScreenSize from '../../../scripts/decorate.js';
 
+export const CSSRanges = {
```
[the remark this answers is not in the document]
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js]: done

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js (hunk 2 above)
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js]: done

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js (hunk 3 above)
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js]: done

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js [hunk 8]
```diff
@@ -199,6 +247,52 @@ function uploadImage(media, layer) {
...
+              colorize: false,
+              presetKind: {
+                _enum: 'presetKindType',
+                _value: 'presetKindCustom',
+              },
+            },
+          },
+        },
+      ];
+      const psurls = [
```
[the remark this answers is not in the document]
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js]: Could you please check if this is the expected functionality

File: creativecloud/features/interactive-components/slider-tray/slider-tray.js (hunk 4 above)
Comment (suhjainadobe) [reply creativecloud/features/interactive-components/slider-tray/slider-tray.js]: Please review it again

**Review by aishwaryamathuria — APPROVED** [review]

**Review by salonijain3 — APPROVED** [review]

**Comment by spadmasa** [comment]
Validation done in feature branch on below urls :
https://openinps--cc--adobecom.hlx.page/drafts/Souj/interactivelements/demo/huesat
image
image

image

Please note that continue to photoshop functionality on redirection to photoshop web will be verified once the pr is merged to stage.
image
image
image

image

image

https://openinps--cc--adobecom.hlx.page/jp/drafts/Souj/interactivelements/demo/huesat
https://openinps--cc--adobecom.hlx.page/fr/drafts/Souj/interactivelements/demo/huesat/hue-sat-marquee
image
image
image

image

image

**Pull request merged**